packages/kbn-config-schema/src/duration/index.test.ts

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { ensureDuration } from '.';
+
+describe('duration', () => {
+  describe('as string', () => {
+    test('ensureDuration 30s', () => {
+      expect(ensureDuration('30s').asMilliseconds()).toEqual(30000);
+    });
+
+    test('ensureDuration 1m30s', () => {
+      expect(ensureDuration('1m30s').asMilliseconds()).toEqual(90000);
+    });
+
+    test('ensureDuration 1m30s70ms', () => {
```

Comment:
A bit silly, but what would happen with input like: `1m30s90m`?

Reply:
my guess is that `1m` and `90m` are added, resulting in `91m30s`

Reply:
I dug deeper into this and found that the closest function I could find that does this is Golang's `ParseDuration`. And they allow any sorting, and repeating units: https://go.dev/play/p/tjL2EOfdDNo.

Maybe it's not that bad after all 😅 
cc @pgayvallet

Reply:
If we must have it, I tried implementing a sorting validation and ended up with this:

```typescript
import { Duration, duration as momentDuration, isDuration } from 'moment';
export type { Duration };
export { isDuration };

const timeFormatRegex = /^(0|[1-9][0-9]*)(ms|s|m|h|d|w|M|y|Y)(.*)$/;

enum TimeUnit {
  ms,
  s,
  m,
  h,
  d,
  w,
  M,
  y,
  Y, // Keeping this for BWC, but Moment prefers lowercase 'y'
}

type TimeUnitString = keyof typeof TimeUnit;

const sortedUnits = Object.keys(TimeUnit).reverse().join('->');

function stringToDuration(text: string, latestUnit?: TimeUnitString): Duration {
  const result = timeFormatRegex.exec(text);
  if (!result) {
    const number = Number(text);
    if (typeof number !== 'number' || isNaN(number)) {
      throw new Error(
        `Failed to parse value as time value. Value must be a duration in milliseconds, or follow the format ` +
          `<count>[ms|s|m|h|d|w|M|y] (e.g. '70ms', '5s', '3d', '1y', '1m30s'), where the duration is a safe positive integer.`
      );
    }
    return numberToDuration(number);
  }

  const count = parseInt(result[1], 10);
  const unit = result[2] as TimeUnitString;
  const rest = result[3];

  if (latestUnit && TimeUnit[latestUnit] < TimeUnit[unit]) {
    throw new Error(
      `Units are expected to follow an order ${sortedUnits} but got ${unit} after ${latestUnit}`
    );
  }

  const duration = momentDuration(count, unit as Exclude<TimeUnitString, 'Y'>); // Moment still supports capital 'Y', but officially (and type-wise), it doesn't.

  if (rest) {
    return duration.add(stringToDuration(rest, unit));
  }
  return duration;
}
```

As you can see from my comments, I noticed that we are passing `Y` to `moment`, when they expect `y`. I added support to `y` while keeping BWC support for `Y` in [`627d5bb` (#187888)](https://github.com/elastic/kibana/pull/187888/commits/627d5bb1cf3180c28eb5f02614e7f463077886c9)

Reply:
> they allow any sorting, and repeating units: 

Yeah, as I said, the implementation is not perfect because it doesn't follow any real ISO norm and we can do stupid things, but I think it's still fine. If the customer wants to do `30s30s` instead of `1m`, I don't see the risk

Reply:
> I don't see the risk

I agree it's not major, but it has potential to surprise users. If there are other tools that do the same stick with simpler implementation makes sense to me!

Reply:
Maybe just a matter of documenting the behaviour as such.

Reply:
Great point about the documentation! Done in [`1faa654` (#187888)](https://github.com/elastic/kibana/pull/187888/commits/1faa65443c87efeb735256b6874d27e4ce496fb9).